[BUG] Crash on bad telemetry

**Describe the bug**
UI became unresponsive, checking logs there were a few sqlite errors.  A restart fixed it.  Logs below. 

**Logs**
```

> meshmonitor@2.4.5 start
> node dist/server/server.js
[INFO] 🔐 Session configuration:
[INFO]    - Cookie secure: false
[INFO]    - Cookie sameSite: lax
[INFO]    - Environment: production
[INFO] ℹ️  OIDC not configured (missing OIDC_ISSUER, OIDC_CLIENT_ID, or OIDC_CLIENT_SECRET)
[ERROR] ❌ Error processing telemetry message: SqliteError: NOT NULL constraint failed: telemetry.value
    at DatabaseService.insertTelemetry (file:///app/dist/services/database.js:968:14)
    at MeshtasticManager.processTelemetryMessageProtobuf (file:///app/dist/server/meshtasticManager.js:942:37)
    at MeshtasticManager.processMeshPacket (file:///app/dist/server/meshtasticManager.js:610:36)
    at MeshtasticManager.processIncomingData (file:///app/dist/server/meshtasticManager.js:312:32)
    at TcpTransport.<anonymous> (file:///app/dist/server/meshtasticManager.js:44:22)
    at TcpTransport.emit (node:events:519:28)
    at TcpTransport.handleIncomingData (file:///app/dist/server/tcpTransport.js:200:18)
    at Socket.<anonymous> (file:///app/dist/server/tcpTransport.js:57:22)
    at Socket.emit (node:events:519:28)
    at addChunk (node:internal/streams/readable:561:12) {
  code: 'SQLITE_CONSTRAINT_NOTNULL'
}
[ERROR] ❌ TCP socket error: read ETIMEDOUT
[ERROR] ❌ TCP transport error: read ETIMEDOUT
[ERROR] ❌ TCP socket error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP transport error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] Reconnection failed: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP socket error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP transport error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] Reconnection failed: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP socket error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP transport error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] Reconnection failed: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP socket error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP transport error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] Reconnection failed: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP socket error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP transport error: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] Reconnection failed: connect EHOSTUNREACH 10.75.0.243:4403
[ERROR] ❌ TCP socket error: read ECONNRESET
[ERROR] ❌ TCP transport error: read ECONNRESET
[ERROR] ❌ Error processing text message: SqliteError: UNIQUE constraint failed: messages.id
    at DatabaseService.insertMessage (file:///app/dist/services/database.js:732:14)
    at MeshtasticManager.processTextMessageProtobuf (file:///app/dist/server/meshtasticManager.js:718:33)
    at MeshtasticManager.processMeshPacket (file:///app/dist/server/meshtasticManager.js:601:36)
    at MeshtasticManager.processIncomingData (file:///app/dist/server/meshtasticManager.js:312:32)
    at TcpTransport.<anonymous> (file:///app/dist/server/meshtasticManager.js:44:22)
    at TcpTransport.emit (node:events:519:28)
    at TcpTransport.handleIncomingData (file:///app/dist/server/tcpTransport.js:200:18)
    at Socket.<anonymous> (file:///app/dist/server/tcpTransport.js:57:22)
    at Socket.emit (node:events:519:28)
    at addChunk (node:internal/streams/readable:561:12) {
  code: 'SQLITE_CONSTRAINT_PRIMARYKEY'
}```

**Environment (please complete the following information):**
* Host OS: Linux
* Docker 2.4.5

--- a/CLAUDE.md
+++ b/CLAUDE.md
@@ -11,3 +11,4 @@ tools to resolve library id and get library docs without me having to explicitly
 - When testing locally, use the docker-compose.dev.yml to build the local code.  Also, always make sure the proper code was deployed once the container is launched.
 - Official meshtastic protobuf definitions can be found at https://github.com/meshtastic/protobufs/
 - When updating the version, make sure you get both the package.json and Helm chart.. and regenerate the package-lock
+- Prior to creating a PR, make sure to run the tests/system-tests.sh to ensure success.

--- a/src/server/meshtasticManager.ts
+++ b/src/server/meshtasticManager.ts
@@ -1094,25 +1094,25 @@ class MeshtasticManager {
         nodeData.airUtilTx = deviceMetrics.airUtilTx;
 
         // Save all telemetry values from actual TELEMETRY_APP packets (no deduplication)
-        if (deviceMetrics.batteryLevel !== undefined) {
+        if (deviceMetrics.batteryLevel !== undefined && deviceMetrics.batteryLevel !== null && !isNaN(deviceMetrics.batteryLevel)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'batteryLevel',
             timestamp, value: deviceMetrics.batteryLevel, unit: '%', createdAt: now
           });
         }
-        if (deviceMetrics.voltage !== undefined) {
+        if (deviceMetrics.voltage !== undefined && deviceMetrics.voltage !== null && !isNaN(deviceMetrics.voltage)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'voltage',
             timestamp, value: deviceMetrics.voltage, unit: 'V', createdAt: now
           });
         }
-        if (deviceMetrics.channelUtilization !== undefined) {
+        if (deviceMetrics.channelUtilization !== undefined && deviceMetrics.channelUtilization !== null && !isNaN(deviceMetrics.channelUtilization)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'channelUtilization',
             timestamp, value: deviceMetrics.channelUtilization, unit: '%', createdAt: now
           });
         }
-        if (deviceMetrics.airUtilTx !== undefined) {
+        if (deviceMetrics.airUtilTx !== undefined && deviceMetrics.airUtilTx !== null && !isNaN(deviceMetrics.airUtilTx)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'airUtilTx',
             timestamp, value: deviceMetrics.airUtilTx, unit: '%', createdAt: now
@@ -1122,19 +1122,19 @@ class MeshtasticManager {
         const envMetrics = telemetry.environmentMetrics;
         logger.debug(`🌡️ Environment telemetry: temp=${envMetrics.temperature}°C, humidity=${envMetrics.relativeHumidity}%`);
 
-        if (envMetrics.temperature !== undefined && envMetrics.temperature !== null) {
+        if (envMetrics.temperature !== undefined && envMetrics.temperature !== null && !isNaN(envMetrics.temperature)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'temperature',
             timestamp, value: envMetrics.temperature, unit: '°C', createdAt: now
           });
         }
-        if (envMetrics.relativeHumidity !== undefined && envMetrics.relativeHumidity !== null) {
+        if (envMetrics.relativeHumidity !== undefined && envMetrics.relativeHumidity !== null && !isNaN(envMetrics.relativeHumidity)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'humidity',
             timestamp, value: envMetrics.relativeHumidity, unit: '%', createdAt: now
           });
         }
-        if (envMetrics.barometricPressure !== undefined && envMetrics.barometricPressure !== null) {
+        if (envMetrics.barometricPressure !== undefined && envMetrics.barometricPressure !== null && !isNaN(envMetrics.barometricPressure)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'pressure',
             timestamp, value: envMetrics.barometricPressure, unit: 'hPa', createdAt: now
@@ -1144,13 +1144,13 @@ class MeshtasticManager {
         const powerMetrics = telemetry.powerMetrics;
         logger.debug(`⚡ Power telemetry: ch1_voltage=${powerMetrics.ch1Voltage}V`);
 
-        if (powerMetrics.ch1Voltage !== undefined) {
+        if (powerMetrics.ch1Voltage !== undefined && powerMetrics.ch1Voltage !== null && !isNaN(powerMetrics.ch1Voltage)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'ch1Voltage',
             timestamp, value: powerMetrics.ch1Voltage, unit: 'V', createdAt: now
           });
         }
-        if (powerMetrics.ch1Current !== undefined) {
+        if (powerMetrics.ch1Current !== undefined && powerMetrics.ch1Current !== null && !isNaN(powerMetrics.ch1Current)) {
           databaseService.insertTelemetry({
             nodeId, nodeNum: fromNum, telemetryType: 'ch1Current',
             timestamp, value: powerMetrics.ch1Current, unit: 'mA', createdAt: now

--- a/src/services/database.test.ts
+++ b/src/services/database.test.ts
@@ -184,8 +184,9 @@ const createTestDatabase = () => {
     }
 
     insertMessage(messageData: DbMessage): void {
+      // Use INSERT OR IGNORE to silently skip duplicate messages
       const stmt = this.db.prepare(`
-        INSERT INTO messages (
+        INSERT OR IGNORE INTO messages (
           id, fromNodeNum, toNodeNum, fromNodeId, toNodeId,
           text, channel, portnum, timestamp, rxTime, createdAt
         ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
@@ -509,6 +510,38 @@ describe('DatabaseService', () => {
 
       const secondPage = db.getMessages(5, 5);
       expect(secondPage).toHaveLength(5);
+    });
+
+    it('should handle duplicate message IDs gracefully', () => {
+      const message = {
+        id: 'msg-duplicate',
+        fromNodeNum: 1,
+        toNodeNum: 2,
+        fromNodeId: '!sender',
+        toNodeId: '!receiver',
+        text: 'Original message',
+        channel: 0,
+        timestamp: Date.now(),
+        createdAt: Date.now()
+      };
+
+      // Insert the message
+      db.insertMessage(message);
+
+      // Try to insert the same message again (should not throw error)
+      expect(() => {
+        db.insertMessage(message);
+      }).not.toThrow();
+
+      // Verify only one message exists
+      const retrieved = db.getMessage('msg-duplicate');
+      expect(retrieved).toBeTruthy();
+      expect(retrieved.text).toBe('Original message');
+
+      // Verify message count is still 1
+      const messages = db.db.prepare('SELECT COUNT(*) as count FROM messages WHERE id = ?')
+        .get('msg-duplicate') as { count: number };
+      expect(messages.count).toBe(1);
     });
   });
 

--- a/src/services/database.ts
+++ b/src/services/database.ts
@@ -952,8 +952,10 @@ class DatabaseService {
 
   // Message operations
   insertMessage(messageData: DbMessage): void {
+    // Use INSERT OR IGNORE to silently skip duplicate messages
+    // (mesh networks can retransmit packets or send duplicates during reconnections)
     const stmt = this.db.prepare(`
-      INSERT INTO messages (
+      INSERT OR IGNORE INTO messages (
         id, fromNodeNum, toNodeNum, fromNodeId, toNodeId,
         text, channel, portnum, timestamp, rxTime, hopStart, hopLimit, replyId, emoji, createdAt
       ) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)

